core/src/main/java/com/linecorp/armeria/server/ServiceRequestContextBuilder.java

```diff
@@ -62,6 +65,26 @@ public void serverStarting(Server server) throws Exception {
         }
     };
 
+    private static final TimeoutTask noopTimeoutTask = new TimeoutTask() {
+        @Override
+        public boolean canSchedule() {
+            return true;
+        }
+
+        @Override
+        public void run() { /* no-op */ }
+    };
+
+    /**
+     * A timeout controller that has been timed-out.
+     */
+    private static final DefaultTimeoutController noopTimedOutController =
+            new DefaultTimeoutController(noopTimeoutTask, ImmediateEventExecutor.INSTANCE);
```

Comment:
Is there any reason that uses ImmediateEventExecutor.INSTANCE unlike `ClientRequestContextBuilder` uses?
Also, can we put this into the superclass to share?

Reply:
> Is there any reason that uses ImmediateEventExecutor.INSTANCE unlike ClientRequestContextBuilder uses?

Oops, I missed it. I changed this based on https://github.com/line/armeria/pull/2740#discussion_r433627772

>  Also, can we put this into the superclass to share?

This is duplicate code, but I intentionally added these to the subclasses. DefaultTimeoutController is an internal class. We can expose it to public or protected.

Reply:
Yeah, it's fine as it is.